File: src/com/leetcode/Medium/FindFirstAndLastPositionOfElementInSortedArray.java
package com.leetcode.Medium;

// 34. Find First and Last Position of Element in Sorted Array
public class FindFirstAndLastPositionOfElementInSortedArray {

    public int[] searchRange(int[] nums, int target) {
        int[] arr = new int[2];
        arr[0] = findFirstIndex(nums, target);
        arr[0] = findLastIndex(nums, target);
        return arr;
    }

    public static int findFirstIndex(int[] nums, int target) {
        int index = -1;
        int low = 0;
        int high = nums.length - 1;
        while (low <= high) {
            int mid = (low + high) / 2;
            if (nums[mid] >= target) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }

            if (nums[mid] == target)
                index = mid;
        }
        return index;
    }

    public static int findLastIndex(int[] nums, int target) {
        int index = -1;
        int low = 0;
        int high = nums.length - 1;
        while (low <= high) {
            int mid = (low + high) / 2;
            if (nums[mid] <= target) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }

            if (nums[mid] == target)
                index = mid;
        }
        return index;
    }

}

File: JavaScript/Medium/FindFirstAndLastPositionOfElementInSortedArray.js
/**
 * 34. Find First and Last Position of Element in Sorted Array
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const searchRange = function(nums, target) {
    let firstPos = findFirstPosition(nums, target);
    if(firstPos == -1)
        return [-1, -1];
    
    let lastPos = findLastPosition(nums, target);
    return [firstPos, lastPos];
};

    
const findFirstPosition = function (arr, target) {
    let low = 0, high = arr.length - 1, index = -1;
    while (low <= high) {
        let mid = Math.floor(low + (high - low) / 2);
        
        if (arr[mid] == target) {
            index = mid;
            high = mid - 1;
        } else if(target < arr[mid]) {
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }
    return index;
}
    
const findLastPosition = function (arr, target) {
    let low = 0, high = arr.length - 1, index = -1;
    while (low <= high) {
        let mid = Math.floor(low + (high - low) / 2);
        
        if (arr[mid] == target) {
            index = mid;
            low = mid + 1;
        } else if(target < arr[mid]) {
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }
    return index;
}